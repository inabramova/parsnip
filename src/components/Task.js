import React from 'react';
import { connect } from 'react-redux';
import { TASK_STATUSES } from '../constants';

export const Task = props => {
  function onStatusChange(e) {
    props.onStatusChange(props.task.id, e.target.value);
  }
  return (
    <div className="task">
      <div className="task-header">
        <div>{props.task.title}</div>
        <div className="select-style">
          <select value={props.task.status} onChange={onStatusChange}>
            {Object.values(TASK_STATUSES).map(status => (
              <option key={status} value={status}>
                {status}
              </option>
            ))}
          </select>
        </div>
      </div>
      <hr />
      <div className="task-body">{props.task.description}</div>
      <div className="task-timer">{props.task.timer}s</div>
    </div>
  );
};

function mapStateToProps(state, ownProps) {
  return {
    task: state.tasks.items[ownProps.taskId],
  };
}

export default connect(mapStateToProps)(Task);
