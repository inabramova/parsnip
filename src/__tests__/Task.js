import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Task from '../components/Task';

Enzyme.configure({ adapter: new Adapter() });

describe('the Task component', () => {
  const task = {
    id: 1,
    title: 'A',
    description: 'a',
    status: 'Unstarted',
    timer: 0,
  };
  const onStatusChange = jest.fn();

  it('should render a status', () => {
    const wrapper = shallow(
      <Task task={task} onStatusChange={onStatusChange} />
    );

    const select = wrapper.find('select').first();
    expect(select.props().value).toEqual('Unstarted');
    select.prop('onChange')({ target: { value: 'In Progress' } });
    expect(onStatusChange).toHaveBeenCalled();
  });

  it('should match the last snapshot', () => {
    const wrapper = shallow(
      <Task task={task} onStatusChange={onStatusChange} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
