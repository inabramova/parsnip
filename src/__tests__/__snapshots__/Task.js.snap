// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Task component should match the last snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
